gf-webapp-admin/src/component/rooms/room/adults/Adults.tsx

```diff
@@ -1,44 +1,32 @@
 import {useState} from "react";
-import {Room} from "src/model/Room/RoomDeprecated";
-import {saveRoom} from "src/service/RoomService";
-import {Button} from "gf-ui-lib/components/Button/Button";
-import styles from "src/component/rooms/room/adults/adults.module.scss";
+import {useRoomContext} from "src/component/rooms/room/roomContext";
 
-interface AdultsProps {
-  room:Room;
-}
 
-export const Adults = (props: AdultsProps) => {
-  const [adults, setAdults] = useState(props.room.adults);
-  const [isEditFieldDisabled, setIsDisabled] = useState(true);
+export const Adults = () => {
 
-  const saveHandler = () => {
-    setIsDisabled(true);
-    props.room.adults = adults;
-    saveRoom(props.room);
-  };
+  const {room, setRoom} = useRoomContext();
+
+  const [adults, setAdults] = useState(room.adults);
+
+  // TODO: onChangeValue repeated in many components. It will become universal when the common input component is implemented.
 
-  const fieldEditHandler = () => {
-    setIsDisabled(false);
+  const onChangeValue = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = Number(e.target.value);
+    setAdults(value);
+    room.adults = value;
+    setRoom(room);
   };
 
   return (
-    <div className={styles.adults}>
+    <div>
       <label>
-        <h4 className={styles.title}>
+        <h4>
```

Comment:
In project many titles? If yes maybe better make a component title and use it everywhere?

Reply:
This problem will be solved when issue #43 is implemented